synedrion/src/cggmp21/protocols/key_init.rs

```diff
@@ -341,24 +332,27 @@ impl<P: SchemeParams, I: Serialize + Ord + Clone + Debug> Round<I> for Round3<P,
         from: &I,
         broadcast_msg: Self::BroadcastMessage,
         _direct_msg: Self::DirectMessage,
-    ) -> Result<Self::Payload, <Self::Result as ProtocolResult>::ProvableError> {
+    ) -> Result<Self::Payload, <Self::Result as ProtocolResult<I>>::ProvableError> {
         let data = self.others_data.get(from).unwrap();
 
         let aux = (&self.context.sid_hash, from, &self.rid);
         if !broadcast_msg.psi.verify(&data.cap_a, &data.cap_x, &aux) {
-            return Err(KeyInitError::R3InvalidSchProof);
+            return Err(KeyInitError {
+                error: KeyInitErrorType::R3InvalidSchProof,
+                phantom: (PhantomData, PhantomData),
```

Comment:
To make it easier to read perhaps we can spell it out like so:
```suggestion
                phantom: (PhantomData::<P>, PhantomData::<I>),
```